sky/spot/recovery_strategy.py

```diff
@@ -130,13 +130,18 @@ def _try_cancel_all_jobs(self):
             return
         try:
             usage_lib.messages.usage.set_internal()
-            sky.cancel(cluster_name=self.cluster_name, all=True)
+            sky.cancel(cluster_name=self.cluster_name,
```

Comment:
We should document:
```
# Note that `sky.cancel()` may not go through for a variety of reasons: (1) head node is preempted; or (2) somehow user programs escape the cancel codepath's kill. The latter is silent and is a TODO. 
#
# For the former, an exception will be thrown, in which case we fallback to terminate_cluster() in the except block below. This is because in the event of recovery on the same set of remaining nodes, we don't want to leave some old job processes running. TODO: This is non-ideal and we should figure out another way to reliably cancel those processes and not have to down the remaining nodes first.
```

Reply:
Or something better :)